### PR TITLE
Switch to optimized text

### LINKED ISSUE #13579: Switch Text.windows.js to use new optimized Text
### Problem Description

Recent Integration switched Text.js to use new optimized TextOptimized.js and removed the gating. Previously you would need to turn on a flag to use TextOptimized and we never turned this flag on. Trying to switch it in the Integration works for Paper but **breaks Text accessibility for Fabric**. Further work will need to be done to switch to the optimized text.

### Link to the integration where the error originated

https://github.com/microsoft/react-native-windows/pull/13554

### Link to commit

https://github.com/facebook/react-native/pull/45317/files

### Forked files

1. Text.windows.js

### Upstream PR

N/A

### Build Configuration

Debug

### PR DESCRIPTION
**Pull request by TatianaKapos**
## Description
Switches to use new optimized Text.js since upstream deprecated the older version!

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why
Resolves #13579

## Changelog
 no

### CODE REVIEW

**Comment by jonthysell** [comment]
Does this PR fix the accessibility in Fabric issues as mentioned in the original bug?

**Comment by TatianaKapos** [comment]
@jonthysell yes! When merging, it overwrote the default accessible value for windows, so added that back in and marked it as a windows change so it doesn't get changed in the future.

**Review by jonthysell — APPROVED** [review]

File: vnext/src-win/Libraries/Text/Text.windows.js [hunk 1]
```diff
@@ -24,105 +22,435 @@ import {NativeText, NativeVirtualText} from './TextNativeComponent';
...
+      const _accessibilityStateDisabled = _accessibilityState?.disabled;
+      const _disabled = disabled ?? _accessibilityStateDisabled;
+
+      const isPressable =
+        (onPress != null ||
+          onLongPress != null ||
+          onStartShouldSetResponder != null) &&
+        _disabled !== true;
+
+      // TODO: Move this processing to the view configuration.
```
Comment (marlenecota) [inline vnext/src-win/Libraries/Text/Text.windows.js:124]: Is there an issue tracking this TODO?

Comment (TatianaKapos) [reply vnext/src-win/Libraries/Text/Text.windows.js:124]: This is from upstream!

File: vnext/src-win/Libraries/Text/Text.windows.js [hunk 2]
```diff
@@ -142,16 +470,14 @@ const TextLegacy: React.AbstractComponent<
     }
 
     return {
-      disabled: !isPressable,
+      disabled: false,
```
Comment (marlenecota) [inline vnext/src-win/Libraries/Text/Text.windows.js:473]: so it's always disabled?

Comment (acoates-ms) [reply vnext/src-win/Libraries/Text/Text.windows.js:473]: Looks like its always enabled.  - Looks like you cannot make text pressable + disabled. I assume this just aligns with upstream.

**Review by acoates-ms — APPROVED** [review]

File: vnext/src-win/Libraries/Text/Text.windows.js (hunk 2 above)
Comment (TatianaKapos) [reply vnext/src-win/Libraries/Text/Text.windows.js:473]: yes this is also from upstream, disabled is defined above as normal on the other Text objects :)

**Pull request merged**